modules/ossm-configuring-distr-tracing-tempo.adoc

```diff
@@ -0,0 +1,94 @@
+////
+
+:_mod-docs-content-type: PROCEDURE
+[id="ossm-configuring-distr-tracing-tempo_{context}"]
+= Configuring the distributed tracing platform (Tempo)
+
+You can expose tracing data to the {TempoName} stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.
+
+The following example configures the `ServiceMeshControlPlane` resource to define an extensionProvider `zipkin` for sending tracing spans to a TempoStack. You can create a TempoStack instance in the `tracing-system` namespace _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
+
+.Prerequisites
+
+You have installed the {TempoOperator} and {SMProductName} Operator in the `openshift-operators` namespace and have created namespaces such as `istio-system` and `tracing-system`.
+
+.Procedure
+
+. Configure the `ServiceMeshControlPlane` resource to define an extension provider:
++
+[source,yaml]
+----
+kind: ServiceMeshControlPlane
+apiVersion: maistra.io/v2
+metadata:
+  name: basic
+  namespace: istio-system
+spec:
+...
+  meshConfig:
+    extensionProviders:
+      - name: tempo
+        zipkin:
+          service: tempo-sample-distributor.tracing-system.svc.cluster.local
+          port: 9411
+  tracing:
+    sampling: 10000
+    type: None <1>
+  version: v2.5
+---
+kind: ServiceMeshMemberRoll
+apiVersion: maistra.io/v1
+metadata:
+  name: default
+  namespace: istio-system
+spec:
+  members:
+    - tracing-system
+----
+<1> The `spec.tracing.type` setting defines a Distributed Tracing deprecated Jaeger instance. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.
++
+[NOTE]
+====
+Create a TempoStack instance _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
+====
++
+
+. Configure the Kiali resource specification to enable a Kiali workload traces dashboard. You can use the dashboard to view tracing query results.
++
+[source,yaml]
+----
+apiVersion: kiali.io/v1alpha1
+kind: Kiali
+...
+spec:  
+  external_services:
+    tracing:
+      enabled: true
+      in_cluster_url: 'http://tempo-sample-query-frontend.tracing-system.svc.cluster.local:16686'
+      url: '[Tempo query frontend Route url]'
+----
++
+
+. Create a TempoStack instance using the Red Hat {TempoOperator} in the `tracing-system` namespace. For more information, see "Installing the distributed tracing platform (Tempo)" in the {product-title} documentation.
++
+
+. Apply a Telemetry custom resource for {SMProductShortName} to start the Tempo provider setting.
++
+[source,yaml]
+----
+apiVersion: telemetry.istio.io/v1alpha1
+kind: Telemetry
+metadata:
+  name: mesh-default
+  namespace: istio-system
+spec:
+  tracing:
+  - providers:
+    - name: tempo
+    randomSamplingPercentage: 100
```

Comment:
Just checking, is the alignment of `randomSamplingPercentage` OK? I was thinking maybe the "r" needs to align with the "n" (this is not because I am familiar with this resource, just weird to see the parameter aligned with the "-"